code/modules/mob/living/living.dm

```diff
@@ -1297,8 +1297,8 @@
 				SSmobs.clients_by_zlevel[new_z] += src
 				for (var/I in length(SSidlenpcpool.idle_mobs_by_zlevel[new_z]) to 1 step -1) //Backwards loop because we're removing (guarantees optimal rather than worst-case performance), it's fine to use .len here but doesn't compile on 511
 					var/mob/living/simple_animal/SA = SSidlenpcpool.idle_mobs_by_zlevel[new_z][I]
-					if (SA)
-						SA.toggle_ai(AI_ON) // Guarantees responsiveness for when appearing right next to mobs
+					if (SA && get_dist(get_turf(src), get_turf(SA)) < MAX_SIMPLEMOB_WAKEUP_RANGE)
```

Comment:
Won't this not even work? `update_z` is only called on z-level transition, so you'd have to transition to a new z-level AND happen to be within mob range for it to wake up the moment you transition to a new z-level.

Doesn't this effectively mean all mobs just sit there, dumbounded until you attack them?

Reply:
hmm, I asumed this would be done post mob transition. I'll review this

Reply:
also no, mobs wake up if you come in range on the idle loop check, and they do a faster check where they do a dist against all mobs in the z level list unless they're on the station z level in which case they do a standard view.

Reply:
Wait did you think this was the only way idle mobs woke up?

Reply:
As far as I know, `toggle_ai(AI_ON)` only ever happens when you attack the mob---maybe when it runs the `ListTargets()`? I can't recall exactly on the latter, but I know `adjustHealth` or whatever calls it.

Reply:
it's part of the hostile mob consider wakeup default proc, and you can implement a custom consider wakeup if you have needs outside of that too.

Reply:
Hmm. This being turned into something that doesn't require an SS but just instantaneously fires the moment a mob enters aggro range would be neat--it'd make the system a bit more responsive---that is to say, without all the `ListTargets` stuff.

Reply:
FYI, you were right in a way, because the mobs were off, not idle, so we do actually need to flick them to IDLE